app/moderation/ModerationPageContent.tsx

```diff
@@ -0,0 +1,1437 @@
+    color: theme.palette.grey[600]
+  },
+  filterCheckbox: {
+    display: 'flex',
+    alignItems: 'center',
+    gap: '4px',
+    fontSize: '12px',
+    cursor: 'pointer',
+    userSelect: 'none',
+    whiteSpace: 'nowrap',
+    '& input': {
+      cursor: 'pointer'
+    }
+  },
+  sectionCheckbox: {
+    display: 'flex',
+    alignItems: 'center',
+    gap: '4px',
+    fontSize: '12px',
+    cursor: 'pointer',
+    userSelect: 'none',
+    whiteSpace: 'nowrap',
+    fontWeight: 400,
+    '& input': {
+      cursor: 'pointer'
+    }
+  },
+  expandedRejectionReason: {
+    '& ul, & ol': {
+      listStyle: 'none',
+      padding: 0,
+      margin: 0
+    },
+    '& li': {
+      marginBottom: '8px'
+    }
+  },
+  expandedGrid: {
+    '@media (max-width: 768px)': {
+      display: 'block !important',
+      '& > div': {
+        marginBottom: '16px',
+        '&:last-child': {
+          marginBottom: 0
+        }
+      }
+    }
+  }
+}));
+
+interface Props {
+  moderatorCommentIds: string[];
+  moderatorCommentsCount: number;
+  moderatorCommentsOffset: number;
+  moderatorPosts: ModerationPost[];
+  activeRateLimits: ActiveRateLimit[];
+  activeRateLimitsCount: number;
+  activeRateLimitsOffset: number;
+  showExpiredRateLimits: boolean;
+  showNewUserRateLimits: boolean;
+  deletedComments: ModerationComment[];
+  deletedCommentsCount: number;
+  deletedCommentsOffset: number;
+  rejectedPosts: ModerationPost[];
+  rejectedPostsCount: number;
+  rejectedPostsOffset: number;
+  rejectedComments: ModerationComment[];
+  rejectedCommentsCount: number;
+  rejectedCommentsOffset: number;
+  postsWithBannedUsers: ModerationPost[];
+  postsWithBannedUsersCount: number;
+  postsWithBannedUsersOffset: number;
+  usersWithBannedUsers: ModerationUser[];
+  usersWithBannedUsersCount: number;
+  usersWithBannedUsersOffset: number;
+  globallyBannedUsers: GloballyBannedUser[];
+  globallyBannedUsersCount: number;
+  globallyBannedUsersOffset: number;
+  showExpiredBans: boolean;
+  limit: number;
+}
+
+export default function ModerationPageContent(props: Props) {
+  const {
+    moderatorCommentIds,
+    moderatorCommentsCount,
+    moderatorCommentsOffset,
+    moderatorPosts,
+    activeRateLimits,
+    activeRateLimitsCount,
+    activeRateLimitsOffset,
+    showExpiredRateLimits,
+    showNewUserRateLimits,
+    deletedComments,
+    deletedCommentsCount,
+    deletedCommentsOffset,
+    rejectedPosts,
+    rejectedPostsCount,
+    rejectedPostsOffset,
+    rejectedComments,
+    rejectedCommentsCount,
+    rejectedCommentsOffset,
+    postsWithBannedUsers,
+    postsWithBannedUsersCount,
+    postsWithBannedUsersOffset,
+    usersWithBannedUsers,
+    usersWithBannedUsersCount,
+    usersWithBannedUsersOffset,
+    globallyBannedUsers,
+    globallyBannedUsersCount,
+    globallyBannedUsersOffset,
+    showExpiredBans,
+    limit,
+  } = props;
+
+  const classes = useStyles(styles);
+  const searchParams = useSearchParams();
```

Comment:
we currently still have `useLocation` - that doesn't do anything in particular with searchParams, so it's probably fine to switch to using `useSearchParams`, but noting that I haven't thought about it